src/psyclone/psyGen.py

```diff
@@ -5140,6 +5148,34 @@ def scope(self, new_scope):
 
         self._scope = new_scope
 
+    def gen_c_definition(self):
```

Comment:
Although not strictly a part of this PR, I've just discovered that pylint complains about self._scope, self._is_input and self._is_output being defined outside the constructor. If it's not too big a deal, please could you fix this (you need only set them to None or something in the constructor).

Reply:
mm.. pylint is quite strict here, the constructor is calling the setter methods for these, which I think is good to not replicate the error checking code. As you suggested I added the `<variable> = None` before calling the setter methods to solve the linting complain.

Reply:
Oh yes, I hadn't spotted that. Thanks for fixing.